### ISSUE TITLE
Error 6x VEGA64 -> CL_MEM_OBJECT_ALLOCATION_FAILURE when calling clEnqueueNDRangeKernel for kernel 0.

### ISSUE THREAD

**Issue by ciclonite** [fireice-uk/xmr-stak-amd#174]
Hi All,
I've problem to run XMR-STAK-AMD on windows 10 with 6GPU AMD VEGA 64...
MB : b250 Mining Expert
RAM : 16GB
OS : Windows 10 pro 64Bit
DRIVER : Blockchain driver latest version.

`"gpu_threads_conf" : [ 
    { "index" : 0, "intensity" : 2016, "worksize" : 8, "affine_to_cpu" : false },
    { "index" : 0, "intensity" : 1600, "worksize" : 8, "affine_to_cpu" : false },
    { "index" : 1, "intensity" : 2016, "worksize" : 8, "affine_to_cpu" : false },
    { "index" : 1, "intensity" : 1600, "worksize" : 8, "affine_to_cpu" : false },
    { "index" : 2, "intensity" : 2016, "worksize" : 8, "affine_to_cpu" : false },
    { "index" : 2, "intensity" : 1600, "worksize" : 8, "affine_to_cpu" : false }, 
    { "index" : 3, "intensity" : 2016, "worksize" : 8, "affine_to_cpu" : false },
    { "index" : 3, "intensity" : 1600, "worksize" : 8, "affine_to_cpu" : false },
    { "index" : 4, "intensity" : 2016, "worksize" : 8, "affine_to_cpu" : false },
    { "index" : 4, "intensity" : 1600, "worksize" : 8, "affine_to_cpu" : false },
    { "index" : 5, "intensity" : 2016, "worksize" : 8, "affine_to_cpu" : false },
    { "index" : 5, "intensity" : 1600, "worksize" : 8, "affine_to_cpu" : false },
	
],`

Error : CL_MEM_OBJECT_ALLOCATION_FAILURE when calling clEnqueueNDRangeKernel for kernel 0.

clinfo:
https://pastebin.com/cAs2F7ST

I've tried everything but i'm unable to resolve...
Any ideas?

**Comment by UnclWish** [comment]
Try to increase amount of virtual memory.

**Comment by ciclonite** [comment]
I've setup 32GB of inital virtual Memory without success..

**Comment by ciclonite** [comment]
Any ideas?

**Comment by kschroed** [comment]
Got same issue with same config on Vega 56. 8GB Ram w/ 32GB virtual memory configured.

**Comment by psychocrypt** [comment]
We released a new version of the miner: https://github.com/fireice-uk/xmr-stak
Please check if the issue still exists with the new version. 
If the problem is not fixed so far please open a new issue in [the issue section of the new reposetory](https://github.com/fireice-uk/xmr-stak/issues)

**Comment by ducho** [comment]
Same issue with 6xVEGA56. App is not able working with more than 4x Vega56 cards.

**Comment by psychocrypt** [comment]
please use the new release https://github.com/fireice-uk/xmr-stak

**Comment by Ephrimee** [comment]
I have the same issue with 5 vegas in https://github.com/fireice-uk/xmr-stak/releases

**Comment by chmarti** [comment]
I'm seeing this when running only 2 Vega 56s. 4GB physical memory, 16 GB virtual memory...

"gpu_threads_conf" : [ 
    { "index" : 0, "intensity" : 1932, "worksize" : 8, "affine_to_cpu" : false },
    { "index" : 0, "intensity" : 1932, "worksize" : 8, "affine_to_cpu" : false },
    { "index" : 1, "intensity" : 1932, "worksize" : 8, "affine_to_cpu" : false },
    { "index" : 1, "intensity" : 1932, "worksize" : 8, "affine_to_cpu" : false },
],

**Comment by plashani** [comment]
I have the same issue, currently running 2 vega 64- I know that I can get up to 2kh/s per card so the configuration below is not that intense to the point where it shouldn't work. 

when running 2 thread on each gpu I get the error message. I have updated to the newest version of xmr-stak but that didnt help.
Increased Virtual memory as well. 

"gpu_threads_conf" : [
{ "index" : 0, "intensity" : 2000 "worksize" : 8, "affine_to_cpu" : false },
{ "index" : 0, "intensity" : 1700, "worksize" : 8, "affine_to_cpu" : false },
{ "index" : 1, "intensity" : 2000, "worksize" : 8, "affine_to_cpu" : false },
{ "index" : 1, "intensity" : 1700, "worksize" : 8, "affine_to_cpu" : false },
],

Can some please help.

**Comment by plashani** [comment]
More important question I forgot to ask is, what causes the error message in general? Is this lack of ram allocation? 
If I reduce the intensity significantly it gets rid of the error but at the same time, the hash rate drops by 50%. 
These cards are designed to high intensity, so I'm confused by this error.

**Comment by chmarti** [comment]
Yes, it would be great if someone could comment on why this error occurs...

**Comment by nicehet** [comment]
I have latest version of app but still the same problem I have 6x Vega 56. Ethereum mining software work fine with 6 vegas 56.
Any new versions coming up with this fix?

**Comment by seminarian** [comment]
I had the same error with rx580's. Upping page file size to 64000 did it for me.

**Comment by bex10110** [comment]
I had the same error with trying only 4 x RX580's. Windows 10 with 4 GB RAM and 12288 MB page file size. I upped the page file size to 24000MB and it is working. I'm adding more cards this week and will see if I have to change that again.

**Comment by Fonta1n3** [comment]
I am having same issue running 4 AMD 56, was running 1 fine then added 3 only to get this error. Any suggestions?

**Comment by Fonta1n3** [comment]
All fixed, it was a virtual memory issue, upped it to 60000 and it runs fine :)

**Comment by branislav94** [comment]
I had the same problem ruining only 3x Vega64 and i had only 20000 free. This solved my problem: 
**LARGE PAGE SIZE:** Let's your miner access the memory it needs.
**Open gpedit.exe**

- If you have Windows 10 Education/enterprise or Professional:
Windows start key  --> type: "gpedit" and hit enter.
- If you have Windows 10 Home (or Single Language Edition), you do have gpedit but it is disabled and you need to enable it.  Hat tip to askvg.com who provides a bat file that will do all this work for you.  
- Go to askvg.com site here 
- Scroll down to "Method 1: Only for Windows 10 Users"
- Follow instructions to download and run bat file.
- Restart your computer per the instructions.
- Windows start key  --> type: "gpedit.msc" and hit enter.

**With gpedit open:**
- Expand "Windows Settings"
- Expand "Security Settings"
- Expand "Local Policies"
- Select "User Rights Assignment"
- In the right side screen, Scroll down and select: "Lock Pages in Memory"
- Select, "Add User or Group"
- Type the user name you use when you log into windows and select, "check names"
If it is an e-mail address you include the @domain.com portion in the name
- It will put an underline under your name when you know it worked.
- Hit "OK" and then close the main local security policy screen.  You need to restart computer before running the miner but don't do it now... a restart will come naturally later.
- The graphic below shows you the screens you will have just seen.  Click the photo to expand it and make it a useful size (the black boxes are where I blacked out my user name).
![gpedit](https://user-images.githubusercontent.com/18179973/44169550-107e7000-a0d5-11e8-9961-423c5227377c.png)

I found all this on http://vega.miningguides.com/#WinSetup
